### ISSUE TITLE
Database ID and Name of product overwritten when importing similar products with CSV

### ISSUE THREAD

**Issue by jointsystems** [PrestaShop/PrestaShop#12944]
<!-
**Describe the bug**

Briggs and Stratton's price list has over 100k items that are in constant flux. When attempting to import products with similar names differing by one number or letter the database overwrites the existing product.For instance one model number is 19J137-0008-F1 whereas there is a sister product with a different sized crankshaft 19J137-0007-F1, When you import either one using the model number for the name it overwrites the other and you end up with the product name of the one just imported but the description and images of the other often both products disappear and cannot be found with a product search after updating the search after the import , We normally use the model number as the ID but when that did not work we tried using the original vendor number and the UPC number as the ID and tried the same changes with the name to no avail?

--
**To Reproduce**
1. Go to '.CSV Import..'
2. Click on '.change...'
3. Scroll down to '.csv file to import of individual product that has name and or ID one character different from another previously imported.'
4. After import is successful go to preferences then  search and update database.
5. Then go to products
6. Enter name of product just imported click edit.
7. In our experience you will find under information the description of a previously imported product with the name and usually reference still indicting a product with the name or ID differing by one character but the name has now been changed to the new product just imported. For Example:  we imported product model number 19J137-0007-F1 using the   model number for the name and ID and it was successfully imported.  After updating search we imported 19J137-0008-F1 using the same procedure.  When we went to the product list and did a search 19J137-0007-F1 could no longer be found.  When we entered 19J137-0008-F1 that appeared but when we hit edit it had the description of 19J137-0007-F1 as well as the prices and images but the name had been changed.  We have multiple prestashops using 1.6.23 where we had the same problem with the same similar products.  You import one and it changes the name of a previous product with a name differing by one character.
![19J137-0007-F1 product disapears](https://user-images.githubusercontent.com/41153050/54456647-a941a080-4735-11e9-829e-49be311b8ec4.PNG)
![19J137-0007-F1 product](https://user-images.githubusercontent.com/41153050/54456648-a941a080-4735-11e9-9aa2-b549c0b4a654.PNG)

**Screenshots**
If applicable, add screenshots or screenrecords to help explain your problem.

**Additionnal information**
PrestaShop version: N/A
PHP version: N/A

**Comment by jointsystems** [comment]
19J137-0007-F1 is imported successfully  and appears in product list but disappears after 19J137-0008-F1 is imported.  19J137-0008-F1 when you go to edit 19J137-0008-F1 it contains UPC of 19J137-0007-F1.  When 19J137-0007-F1 is reimported using long vendor part # as name when searched in products using ID when edit is chosen ID 19J137-0007-F1 now appears with 19J137-0008-F1 name.
![19J137-0007-F1 No records found](https://user-images.githubusercontent.com/41153050/54478018-23c8f980-47e4-11e9-884d-5e7464cd08a8.PNG)
![19J137-0007-F1 product list](https://user-images.githubusercontent.com/41153050/54478019-23c8f980-47e4-11e9-909c-9906033b5293.PNG)
![19J137-0007-F1 ID now with 19J137-0008-F1 Name](https://user-images.githubusercontent.com/41153050/54478309-0cd7d680-47e7-11e9-9567-169676877e07.PNG)

**Comment by khouloudbelguith** [comment]
Hi @jointsystems,

I did not manage to reproduce the issue with PS1.7.5.1.
I attached a video record.
https://drive.google.com/file/d/1Du_EsWTTbRsAMcWHcdtk4LmDgdmoe7BP/view
As you may already know, normal maintenance support for PrestaShop 1.6 ended in October 2018, which is now in [extended maintenance mode until June 30, 2019](https://www.prestashop.com/en/blog/maintenance-extension-prestashop-1-6).
During this period, and in order to ensure the stability of current 1.6 shops, only critical bugs and security fixes will be fixed.
I close the issue, feel free to open a new one when needed.
Thanks for your understanding!

**Comment by jointsystems** [comment]
I didn’t have any problem importing the two items both said they were completed successfully however when you searched in products for the first one after importing the second one the second one comes up rather than the first which disappeared.  Your video does not show the two products existing together.

 

Why don’t you try it with the original files attached.

**Comment by khouloudbelguith** [comment]
Hi @jointsystems,

>Why don’t you try it with the original files attached.

Sorry, but I did not have any attached files in this issue.
You need to login to GitHub & upload your files.
Thanks!

**Comment by jointsystems** [comment]
We just imported 19N132-0055-F1 and it overwrote 19J137-0007-F1.  As you can see from the screenshot below the name 19N132-0055-F1 now comes up in a search with the description of 19J137-0007-F1 and 19J127-0008-F1 because we could not separate  0007 and 0008. If you search for either 19J137-0007 or 0008-F1 neither comes up.

**Comment by khouloudbelguith** [comment]
Hi @jointsystems,

I just tried with another shop & it is OK with PS1.7.5.1.
https://drive.google.com/file/d/1FZWU6oc4wrPhtMHwdV2PWiCK6Qaxm690/view
Thanks!

**Comment by jointsystems** [comment]
Here is an image of the 19N132-0055-F1 import and the search results where it overwrites a previous import with both a different name and ID.

![19N132-0055-F1 overwrite](https://user-images.githubusercontent.com/41153050/54821619-1c3ca100-4c79-11e9-9d8e-bd3787e44671.PNG)

![19N132-0055-F1 csv](https://user-images.githubusercontent.com/41153050/54822009-3925a400-4c7a-11e9-8565-dc6bd96cb3cd.PNG)

**Comment by khouloudbelguith** [comment]
Hi @jointsystems,

Could you please send me your file CSV to test it.

Thanks!

**Comment by jointsystems** [comment]
When we put in 19N132-0035 it imports fine, and shows up in product search.  Then you import 19N132-0055-F1, which also works fine and shows up in a product search but the 0035 no longer does but some of the data and the pictures and full description (added after import) are for the 0035.  

 

We don’t use 1.7 as last several times we attempted to upgrade numerous modules did not work the most import being authorize used for processing payments.  Also the ups and Fedex modules were  not supported by 1.7.

**Comment by jointsystems** [comment]
![15T212-0160-F8 over written by 15C112-3007-F8](https://user-images.githubusercontent.com/41153050/54854549-ac0a3b80-4cc9-11e9-8246-eb0a78fdde48.PNG)
After importing 15T212-0160-F8 it came up fine in the product search then we imported 15C112-3007 as it had been overwritten by 15T212-0160-F8.  In the attached image you will note that we searched by the ID for 15T212-0160-F8 what comes up is the name for 15C112-3007 as well as the reference,

**Comment by khouloudbelguith** [comment]
Hi @jointsystems,

I did not manage to reproduce the issue with PS:1.6.1.23 and PS:1.7.5.1
I attached a video record.
https://drive.google.com/file/d/1krOPfDSxNKf96gMSBZlnVfHGSSsC9MC1/view

Thanks!

**Comment by khouloudbelguith** [comment]
@jointsystems, As you may already know, normal maintenance support for PrestaShop 1.6 ended in October 2018, which is now in [extended maintenance mode until June 30, 2019](https://www.prestashop.com/en/blog/maintenance-extension-prestashop-1-6).
Thanks for your understanding!

**Comment by jointsystems** [comment]
I mistakenly upgraded to 1.7.5.2 from 1.6.2.3 using 1 click upgrade and now the shop is unusable.  The time out in the dashboard goes around endlessly.  When you go into manage modules you get a 500 server error.

Could not find 1-click upgrade to roll back the shop, so I used the browser backup till I came back to upgrade page that was now on 1.7.5. 2 but  when you clicked on maintenance and rollback nothing happened?  There does not seem to be any tab for 1-click upgrade?
